zap/src/main/java/org/zaproxy/zap/authentication/AuthenticationMethodType.java

```diff
@@ -214,6 +218,8 @@ public String toString() {
      */
     public abstract ApiDynamicActionImplementor getSetCredentialsForUserApiAction();
 
+    public abstract void replaceUserDataInPollRequest(HttpMessage msg, User user);
```

Comment:
Should this be in the method not the type? It would avoid creating a new type each time the data needs to be replaced.